apps/src/StudioApp.js

```diff
@@ -498,6 +496,22 @@ StudioApp.prototype.init = function (config) {
   this.emit('afterInit');
 };
 
+StudioApp.prototype.initProjectTemplateWorkspaceIconCallout = function () {
```

Comment:
I would push all of this down into the `ProjectTemplateWorkspaceIcon` component, in a [`componentDidMount` lifecycle method](https://reactjs.org/docs/react-component.html#componentdidmount).

Pro: This logic gets colocated with the component it has to attach to, and can safely run any time that component renders so there's no duplicated check for whether we are showing the icon.  Also minimizes how much we're adding to StudioApp.js.

Con: Adds an external dependency and side-effect to what is currently a pure display component.

Thoughts?

Reply:
I can see both perspectives, but given that I'm needing to do special handling of this for weblab (see discussion below), I think it is best to leave it as is.